File: backend/SpringBoot2-Java/src/test/java/com/weweb/auth/service/InMemoryUserServiceTest.java
package com.weweb.auth.service;

import static org.junit.Assert.*;

import com.ufoscout.coreutils.validation.SimpleValidatorService;
import org.junit.Test;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

public class InMemoryUserServiceTest {

    InMemoryUserService service = new InMemoryUserService(new BCryptPasswordEncoder(), new SimpleValidatorService());

    @Test
    public void shouldReturnUser() {
        User user = service.login("user", "user");
        assertNotNull(user);
        assertEquals("user", user.getUsername());
        assertNotEquals("user", user.getEncodedPassword());
    }

    @Test(expected = BadCredentialsException.class)
    public void shouldThrowBadCredentialException() {
        service.login("user", "admin");
    }

}

File: backend/Vertx3-Kotlin/src/test/java/com/weweb/auth/service/InMemoryUserServiceTest.kt
package com.weweb.auth.service

import com.ufoscout.coreutils.validation.SimpleValidatorService
import com.ufoscout.vertk.kodein.auth.BadCredentialsException
import com.weweb.BaseTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

class InMemoryUserServiceTest: BaseTest() {

    internal var service = InMemoryUserService(BCryptPasswordEncoder(), SimpleValidatorService())

    @Test
    fun shouldReturnUser() {
        val user = service.login("user", "user")
        assertNotNull(user)
        assertEquals("user", user.username)
    }

    @Test
    fun shouldThrowBadCredentialException() {
        assertThrows<BadCredentialsException> {
            service.login("user", "admin")
        }
    }

}